mindmeld/converter/dialogflow.py

```diff
@@ -0,0 +1,375 @@
+            new_dict['cname'] = item['value']
+            mapping_dict["entities"].append(new_dict)
+
+            target_gazetteer.write(item['value'] + "\n")
+
+        json.dump(mapping_dict, target_mapping, ensure_ascii=False, indent=2)
+
+        source_en.close()
+        target_gazetteer.close()
+        target_mapping.close()
+
+    def _create_intents_directories(self, intents):
+        """ Creates directories + files for all languages/files."""
+
+        for languages in intents.values():
+            for language, sub in languages.items():
+                dialogflow_intent_file = os.path.join(self.dialogflow_project_directory,
+                                                      "intents", sub + ".json")
+
+                mindmeld_intent_directory_name = self.clean_check(sub, self.intents_list)
+                mindmeld_intent_directory = os.path.join(self.mindmeld_project_directory,
+                                                         "domains", "general",
+                                                         mindmeld_intent_directory_name)
+
+                self.create_directory(mindmeld_intent_directory)
+
+                self._create_intent_file(dialogflow_intent_file,
+                                         mindmeld_intent_directory,
+                                         language)
+
+    def _create_intent_file(self, dialogflow_intent_file, mindmeld_intent_directory, language):
+        source_en = open(dialogflow_intent_file, 'r')
+        target_test = open(os.path.join(mindmeld_intent_directory, "test.txt"), 'w')
+        target_train = open(os.path.join(mindmeld_intent_directory, "train.txt"), 'w')
+
+        datastore = json.load(source_en)
+        all_text = []
+
+        for usersay in datastore:
+            sentence = ""
+            for texts in usersay["data"]:
+                df_text = texts["text"]
+                if "meta" in texts and texts["meta"] != "@sys.ignore":
+                    df_meta = texts["meta"]
+
+                    if re.match("(@sys.).+", df_meta):  # if text is a dialogflow sys entity
+                        if df_meta in DialogFlowConverter.sys_entity_map:
+                            mm_meta = DialogFlowConverter.sys_entity_map[df_meta]
+                        else:
+                            mm_meta = "[DNE: {sysEntity}]".format(sysEntity=df_meta[1:])
+                            logger.info("Unfortunately mindmeld does not currently support"
+                                        "%s as a sys entity."
+                                        "Please create an entity for this.", df_meta[1:])
+
+                        intent = self.clean_name(mm_meta) + "_entries_" + language
+                        part = "{" + df_text + "|" + intent + "}"
+                    else:
+                        intent = self.clean_name(df_meta[1:]) + \
+                                                        "_entries_" + language
```

Comment:
Have we tested the generated mindmeld app on intents created like this, example `greeting_de`? I dont think this would work since the classifier will be highly confused between the language patterns of various languages since this format assumes they live in the same space for the classifier to infer. Please remove language support from this PR and add that as a TODO once MindMeld supports languages.

Reply:
I think Alan will know the answer to this question.

Reply:
@vijay120 : we have tested the DF converter code with different languages, and it works out okay. I think it's okay to keep them here for now until we update MindMeld with a better internationalization model.